airflow/models/dag.py

```diff
@@ -1753,6 +1756,44 @@ def get_last_dagrun(self, session=None, include_externally_triggered=False):
     def safe_dag_id(self):
         return self.dag_id.replace('.', '__dot__')
 
+    @property
+    def code(self):
```

Comment:
The property that invokes a database query is very dangerous.  This can result in a mistake that has a big impact on performance.
Here is an example:
https://github.com/apache/airflow/pull/7476
The property that calls the database query caused the scheduler to run 33% slower.  While reviewing the code, we never could see this problem.

Reply:
Sure, it is a bad practice in any language to have heavy processing in something that seems to be a getter.
My question is, can you somehow cache value as for example _source_code in a class that represents a table in sql alchemy and whatever we have on top of it?